File: typescript/arrays/max.ts
// maximum number in an array

const arr = [1, 2, 3, 4, 5];


function maxNum(arr: number[]): number {

    if(arr.length === 0) {
        return -1;
    }
    let maxVal: number = arr[0];
    for(const i of arr) {
        if(i > maxVal) {
            maxVal = i;
        }
    }
    return maxVal;


}

console.log(maxNum(arr));

function max(arr: number[]): number {
    return Math.max(...arr);
    }

console.log(max([1, 2, 3, 4, 5]));

File: golang/arrays/max.go
package main

import (
	"fmt"
)

func max(arr []int) int {
	max := arr[0]
	for _, val := range arr {
		if val > max {
			max = val
		}
	}
	return max
}

func maxRange(arr []int, start int, end int) int {
	if start > end {
		return -1

	}

	if arr == nil {
		return -1
	}

	maxVal := arr[start]
	for i := start; i <= end; i++ {
		if arr[i] > maxVal {
			maxVal = arr[i]
		}
	}

	return maxVal

}

func main() {
	arr := []int{1, 3, 2, 18, 3, 4, 65}
	fmt.Print(max(arr))
	fmt.Println()
	fmt.Print(maxRange(arr, 0, 3))

}
